### DISCUSSION TITLE
What happened to the `if-none-match` header ? How to use ETag with Remix ?

### DISCUSSION THREAD

**Discussion by kael** [remix-run/remix#5710] [Q&A]
I'm trying to implement cache management using `ETag` but the [`if-none-match` header is vanished](https://github.com/remix-run/remix/blob/ca382b16223e2766fe333b22f177550d2f9070fc/packages/remix-cloudflare-pages/worker.ts#L53-L54) apparently due to [a bug with Cloudflare pages](https://github.com/cloudflare/workers-sdk/issues/117).

I can't see the `if-none-match` header in the `entry.server.tsx` `handleRequest/request` argument nor in the module `headers#headers` argument.

This makes the [use of `ETag`](https://sergiodxa.com/articles/use-etags-in-remix) impossible, doesn't it ? Or I'm missing something ? :thinking: 

My goal is to handle cache management with `Etag` at the `loader` level by comparing the `ETag` returned by the API with the `if-none-match` header of the request.

**Comment by kiliman** [comment]
This was what I posted in Discord last year about Cloudflare and Etags.

>Finally figured out why content-type html wasn't sending etag header... it's because Cloudflare was automatically adding the analytics script to the page. Since this technically altered the response, Cloudflare wisely chose to remove the etag since the content had changed. By removing the auto-setup of analytics, Cloudflare no longer modified the page and now I see etags! Yay!!

>So it would be nice if this article would be updated to also include disable auto-setup of analytics

<img width="1080" alt="image" src="https://user-images.githubusercontent.com/47168/223835571-be9fd9bc-df91-4ede-8fb5-36253283b89d.png">

https://support.cloudflare.com/hc/en-us/articles/218505467-Using-ETag-Headers-with-Cloudflare

https://discord.com/channels/770287896669978684/933780834689880144/934105177751961600
